src/elm/external.ts

```diff
@@ -22,21 +25,49 @@ export class Retrieve extends Expression {
   }
 
   exec(ctx: Context) {
-    let records = ctx.findRecords(this.templateId != null ? this.templateId : this.datatype, this);
-    let codes = this.codes;
-    if (this.codes && typeof this.codes.exec === 'function') {
-      codes = this.codes.execute(ctx);
-      if (codes == null) {
+    // Object with retrieve information to pass back to patient source
+    const retrieveDetails: RetrieveDetails = {
+      datatype: this.datatype
+    };
+
+    let resolvedCodes: Code[] | ValueSet | undefined;
+    if (this.codes) {
+      resolvedCodes = this.codes.execute(ctx);
+
+      if (resolvedCodes == null) {
         return [];
       }
-    }
-    if (codes) {
-      records = records.filter((r: any) => this.recordMatchesCodesOrVS(r, codes));
+
+      retrieveDetails.codes = resolvedCodes;
     }
     // TODO: Added @dateProperty check due to previous fix in cql4browsers in cql_qdm_patient_api hash: ddbc57
+    let range: Interval | undefined;
     if (this.dateRange && this.dateProperty) {
```

Comment:
We checked for both `dateRange` and `dateProperty` in our code because we really needed both for it to be useful at all.  BUT... I think that for the purposes of passing this through to the provider, we shouldn't make such assumptions.

In other words, I'd suggest removing `&& this.dateProperty` from the condition.